File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        int min_price= INT_MAX;
        int max_profit= 0;
        for(int i = 0; i < p.size(); i++){
          if(p[i] < min_price){
            min_price = p[i];
          }
          else if(p[i] - min_price > max_profit){
            max_profit = p[i] - min_price;
          }
        }
        return max_profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minprice = prices[0];
        int maxprof = 0;
        for (int i = 0; i < prices.length; i++){
        if(minprice > prices[i]){
            minprice = prices[i];
        }
        int prof = prices[i] - minprice;
        maxprof = Math.max(prof,maxprof);
        }
        return maxprof;
    }
}